mypy/server/astdiff.py

```diff
@@ -209,3 +135,210 @@ def visit_type_type(self, left: TypeType) -> bool:
         if isinstance(self.right, TypeType):
             return is_identical_type(left.item, self.right.item)
         return False
+
+
+SnapshotItem = Tuple[object, ...]
```

Comment:
It would be nice if you gave at least some examples of what various snapshots look like. Without knowing the internal structure of snapshots the code below is hard to follow. The representation may be easy to manipulate, but it seems to hide logical structure.

Longer term this should probably become a class rather than just a tuple.